go/pkg/casng/streaming_upload.go

```diff
@@ -128,7 +129,61 @@ func (u *StreamingUploader) Upload(ctx context.Context, in <-chan UploadRequest)
 // streamPipe is used by both the streaming and the batching interfaces.
 // Each request will be enriched with internal fields for control and logging purposes.
 func (u *uploader) streamPipe(ctx context.Context, in <-chan UploadRequest) <-chan UploadResponse {
-	panic("not yet implemented")
+	ch := make(chan UploadResponse)
+
+	// If this was called after the the uploader was terminated, short the circuit and return.
```

Comment:
It's intentional because the termination sequence is ordered to ensure every concurrent worker gets the expected messages.
If `u.ctx` is cancelled mid-flight, gRPC calls and file system walks would be cancelled and short-circuited, relevant public methods would also be short-circuited to stop inject more requests, and the whole pipeline is drained to allow each worker to terminate gracefully.

For this particular method, it means that all of in-flight requests will be returned to the pipe, and incoming requests will still be accepted but short-circuited along the way and returned quickly. This is necessary to ensure the caller doesn't deadlock sending on an abandoned channel. In other words, the caller owns the input channel and must close it before this method can terminate. This is mentioned in the godocs of public methods.